### ISSUE TITLE
Maximum message size we can publish to chronicle using low level API

### ISSUE THREAD

**Issue by skaminen** [OpenHFT/Chronicle-Queue#522]
Hi ,

creating a new issue to give visibility. 

Looks like when we configure Hourly roll, the message size is 1/4 of your block size is resulting in core dump. Below is a modification of above code that results in core dumped.

```
public class CQAppenderTest2 {

static final int ITER = 100;
static final int SIZE = 64 * 1024;
static final int BLOCK_SIZE =  SIZE*4;

/**
 * @param args
 */
public static void main(String[] args) {
	// TODO Auto-generated method stub
	
	ChronicleQueue queue = ChronicleQueueBuilder.single("queueDir")
			.blockSize(BLOCK_SIZE)
			.rollCycle(RollCycles.HOURLY) // defining hourly roll causes this code to core dump.
			.wireType(WireType.FIELDLESS_BINARY)
			.build();
	final ExcerptAppender appender = queue.acquireAppender();
	byte[] array = new byte[SIZE];
	long startTime = System.currentTimeMillis();
	for (int i = 0; i < ITER; i++) {
		appender.writeBytes(b -> {
		    long address = b.address(b.writePosition());
		    Unsafe unsafe = UnsafeMemory.UNSAFE;
		    unsafe.copyMemory(array, Unsafe.ARRAY_BYTE_BASE_OFFSET, null, address, array.length);
		    b.writeSkip(array.length);
		});
	}
	long estimatedTime = System.currentTimeMillis() - startTime;
	System.out.println("time = " + estimatedTime);

}
}

```
`18:55:49.814 [main] DEBUG net.openhft.chronicle.bytes.MappedFile - Allocation of 0 chunk in queueDir/20180802-23.cq4 took 6.373 ms.

A fatal error has been detected by the Java Runtime Environment:

SIGSEGV (0xb) at pc=0x000000010e0dc549, pid=2880, tid=0x0000000000001c03

JRE version: Java(TM) SE Runtime Environment (8.0_171-b25) (build 1.8.0_171-b25)

Java VM: Java HotSpot(TM) 64-Bit Server VM (25.171-b25 mixed mode bsd-amd64 compressed oops)

Problematic frame:

V [libjvm.dylib+0xdc549] acl_CopyRight+0x42

Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

#`

Stack trace

Stack: [0x0000700009aca000,0x0000700009bca000], sp=0x0000700009bc9618, free space=1021k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
V [libjvm.dylib+0xdc549] acl_CopyRight+0x42
j sun.misc.Unsafe.copyMemory(Ljava/lang/Object;JLjava/lang/Object;JJ)V+0
j com.CQAppenderTest2.lambda$0([BLnet/openhft/chronicle/bytes/BytesOut;)V+30
j com.CQAppenderTest2$$Lambda$18.writeMarshallable(Lnet/openhft/chronicle/bytes/BytesOut;)V+5
j net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreAppender.writeBytes(Lnet/openhft/chronicle/bytes/WriteBytesMarshallable;)V+19
j com.CQAppenderTest2.main([Ljava/lang/String;)V+56
v ~StubRoutines::call_stub

**Comment by dpisklov** [comment]
@skaminen why are you trying to use Unsafe?
If you use normal API it just works:
```
public static void main(String[] args) {
        // TODO Auto-generated method stub

        ChronicleQueue queue = ChronicleQueueBuilder.single("queueDir")
                .blockSize(BLOCK_SIZE)
                .rollCycle(RollCycles.HOURLY) // defining hourly roll causes this code to core dump.
                .wireType(WireType.FIELDLESS_BINARY)
                .build();
        final ExcerptAppender appender = queue.acquireAppender();
        byte[] array = new byte[SIZE];
        long startTime = System.currentTimeMillis();
        for (int i = 0; i < ITER; i++) {
            appender.writeBytes(b -> b.write(array));
        }
        long estimatedTime = System.currentTimeMillis() - startTime;
        System.out.println("time = " + estimatedTime);

    }
```
Direct write bypasses Chronicle's offset checks so you're shooting yourself in the foot.

**Comment by skaminen** [comment]
@dpisklov Thanks for the feedback. 

The low level api is documented in the Chronicle docs. In our case we were copying from a mix of directMemory and array. The low level API would be useful to integrate with Aeron package  See Fragment Handler API. - https://github.com/real-logic/aeron/blob/master/aeron-client/src/main/java/io/aeron/logbuffer/FragmentHandler.java 

We did make the capacity check with - int capacity = (int) (out.writeLimit() - out.writePosition()) ; , But that didn't protect us in this case.
